game-core/src/system/ally.rs

```diff
@@ -21,62 +21,48 @@ pub struct Grouper;
 
 impl<'s> System<'s> for Grouper {
```

Comment:
I think most of this code should be moved to a separate system. In my mind Grouper is for deciding if an ally should start following a player. The majority of this logic belongs an ally::Movement system, to follow conventions I've been using throughout the rest of the project.